mj_envs/hand_manipulation_suite/relocate_v0.py

```diff
@@ -78,8 +78,8 @@ def gs(self):
         qp = self.data.qpos.ravel().copy()
         qv = self.data.qvel.ravel().copy()
         hand_qpos = qp[:30]
-        obj_pos  = self.data.body_xpos[self.obj_bid].ravel()
-        palm_pos = self.data.site_xpos[self.S_grasp_sid].ravel()
+        obj_pos  = self.model.body_pos[self.obj_bid].ravel()
+        palm_pos = self.model.site_pos[self.S_grasp_sid].ravel()
```

Comment:
model.body_pos provides the position wrt the body frames. 
data.body_xpos provides the positions wrt the world frame.

In information needed here are wrt the world frame.  I, therefore, believe that the original code is correct. Thoughts?

Reply:
The problem is, that in gs we get position wrt to the world frame and in ss we set it wrt to the parent frame, hence mismatch when reading saved state.

Reply:
Since the parent frame of both the object and the target is the world, there should be no difference between the two. As a general rule of thumb, one should read info off mjData. If there is a need to reset the simulation, that should be done only via qpos, qvel, act, and parameters of mjModel (as the rest of mjData is get updated by the mujoco core).

While writing the response, I realized that `palm_pos` isn't a part of the state and should be removed from the gs() function.  Here is an updated version of the two functions:
    def gs(self):
        """
        Get state of hand as well as objects and targets in the scene
        """
        qp = self.data.qpos.ravel().copy()
        qv = self.data.qvel.ravel().copy()
        hand_qpos = qp[:30]
        obj_pos  = self.data.body_xpos[self.obj_bid].ravel()
        target_pos = self.data.site_xpos[self.target_obj_sid].ravel()
        return dict(hand_qpos=hand_qpos, obj_pos=obj_pos, target_pos=target_pos, qpos=qp, qvel=qv)

    def ss(self, state_dict):
        """
        Set the state which includes hand as well as objects and targets in the scene
        """
        qp = state_dict['qpos']
        qv = state_dict['qvel']
        obj_pos = state_dict['obj_pos']
        target_pos = state_dict['target_pos']
        self.set_state(qp, qv)
        self.model.body_pos[self.obj_bid] = obj_pos
        self.model.site_pos[self.target_obj_sid] = target_pos
        self.sim.forward()

Does it adequately address the discrepancies at your end?

Reply:
This makes naturally a lot of sense, but for some reason it did not work correctly for me previously even though it clearly should. I am a bit busy right now, but I will come back to it soon and confirm the proposed change. Thank you for your current input!

Reply:
I have updated the get_state and set_state functions in my latest PR. It has a different naming convention (consistent with trajopt here: https://github.com/aravindr93/trajopt). In the checks that I ran, it seems to work pretty well.

Also worth noting that there are some changes across mujoco 1.5 vs 2.0. Might be easier if we start from that PR and incorporate any fixes necessary. @Jendker can you try out the latest redesign branch and see if that works for you? (this has to be used in conjunction with the redesign branch in other related repos like mjrl and hand_dapg). Thanks so much for helping with the repo!